lib/narn-lib.js

```diff
@@ -22,45 +39,69 @@ exports.detectYarn = function detectYarn(args) {
 
 exports.getYarnArgs = narnArgs => narnArgs;
 
-exports.getNpmArgs = narnArgs => {
-  const yarnArgs = minimist(narnArgs, { boolean: ["dev", "D"] });
+exports.getNpmArgs = (narnArgs, isPnpm = false) => {
+  const yarnArgs = minimist(narnArgs, {
+    boolean: ["dev", "D"],
+    alias: {
+      D: "dev"
+    }
+  });
   const yarnCommands = yarnArgs._;
   const yarnTarget = yarnCommands.length > 0 ? yarnCommands[0] : "install";
   const yarnSubCommands = yarnCommands.slice(1);
+  const narnSubArgs = narnArgs.slice(1);
+
   let npmTarget;
   let npmArgs;
   let result;
   switch (yarnTarget) {
     case "install":
       npmTarget = "install";
-      npmArgs = [];
+      npmArgs = narnSubArgs;
       break;
     case "add":
       npmTarget = "install";
       const packages = yarnSubCommands.map(transformNarnPackageString);
-      const saveType = yarnArgs.dev || yarnArgs.D ? "--save-dev" : "--save";
+      const saveType = yarnArgs.dev ? "--save-dev" : "--save";
       npmArgs = [saveType, ...packages];
       break;
     case "remove":
       npmTarget = "uninstall";
-      npmArgs = yarnSubCommands;
+      npmArgs = narnSubArgs;
       break;
     case "init":
       npmTarget = "init";
-      npmArgs = [];
+      npmArgs = narnSubArgs;
+      break;
+    case "update":
+    case "upgrade":
+      npmTarget = "update";
+      npmArgs = narnSubArgs;
```

Comment:
Again, not all flags to yarn are the same as to npm. If we want to handle flags, we should handle them individually

Reply:
I already commented this thread on the previous one.

The "update" command is literally the same on any implementation.

[npm update](https://docs.npmjs.com/cli-commands/update.html)
[pnpm update](https://pnpm.js.org/en/cli/update)
[yarn upgrade](https://classic.yarnpkg.com/en/docs/cli/upgrade)

Reply:
About the flags, it's completely possible to handle them individually AND pass them around, with the implementation and made one doesn't kill the other.

Reply:
What isn't the same is that the name is different. `yarn update` is not a valid command.

https://github.com/joeldenning/narn/pull/13#discussion_r376681334